### PR TITLE
Allow returning subsampled alignments in PythonControl

### PR DESCRIPTION
**Pull request by NeoLegends**
Currently `PythonControl` requires the number of targets it gives to RETURNN matches the number of features. This is a safety assertion to catch data errors. However, it is wrong when training subsampling models that use different time steps for the input features and output labels.

The PR introduces a parameter that sets the subsampling ratio (e.g. 3x, 4x, etc.) and adapts the length validation assertion to respect the subsampling factor accordingly.

### CODE REVIEW

**Review by Marvin84 — APPROVED** [review]

**Comment by SimBe195** [comment]
Maybe the assertion can also simply be removed altogether. Training would crash later anyway if the time axis of logits and targets don't match up. I would be fine with getting rid of such additional checks.

**Comment by NeoLegends** [comment]
I don't mind either way.

I agree that RETURNN will also catch issues in the data lengths and that the assertion here is more of a defense-in-depth than really necessary.

I would like not to spend too much time on reviewing this PR, so if you prefer I'll completely remove the assertion.

**Comment by Marvin84** [comment]
I suggest we remove it.

**Review by Marvin84 — APPROVED** [review]

**Review by SimBe195 — APPROVED** [review]
